### DISCUSSION TITLE
2012/06/13/using-c-5-caller-info-attributes-when-targeting-earlier-versions-of-the-net-framework/

### DISCUSSION THREAD

**Discussion by giscus** [thomaslevesque/blog#111] [Blog comments]
# 2012/06/13/using-c-5-caller-info-attributes-when-targeting-earlier-versions-of-the-net-framework/

https://thomaslevesque.com/2012/06/13/using-c-5-caller-info-attributes-when-targeting-earlier-versions-of-the-net-framework/

**Comment by landonre** [comment]
I know you wrote this years ago, but it made my life much easier today!  Thank you!!!  

I am having to retrofit several of my completed components to function with Framework 4.0 because we have some older equipment and vendor supplied applications which they cannot upgrade to any newer windows versions.  Those vendor apps and equipment are tied to Windows XP (its only 25 years old (said sarcastically)).   

We are updating our plant database and the database connectivity my components provide to those vendor apps had to be updated to later Oracle clients.  I completed that work 13 months ago when I was assured they would have everything working on Windows 11 shortly.  Well two months from our database upgrade, then are not even close to having it work, so I have to make my components work with XP.  

I had been using compiler directives to redefine all methods where I used the [Caller..}. attributes because these components are used by most of my new applications where I want that functionality and only a few apps require Net 4.  I found your hack today and it works perfectly so far, so I am not removing all of the no longer needed compiler directives.

It may have seemed a small suggestion to you years ago, but if has really helped me with these last-minute modifications I must complete and is much appreciated!!!

**Reply by thomaslevesque** [reply]
Glad it helped!
